sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HostPolicy.java

```diff
@@ -39,7 +39,7 @@ public HostPolicy(String host) {
             context.getHttpRequest().setUrl(urlBuilder.setHost(host).toURL());
             result = next.process();
         } catch (MalformedURLException e) {
-            result = Mono.error(e);
+            result = Mono.error(new RuntimeException("Please check the host URL: " + urlBuilder.setHost(host), e));
```

Comment:
Same here. Error message shouldn't include corrective actions. "Host URL is invalid" would be better.